codegenerator/cli/templates/static/codegen/src/Index.res

```diff
@@ -1,46 +1,153 @@
-/**
- * This function can be used to override the console.log (and related functions for users). This means these logs will also be available to the user
- */
-let overrideConsoleLog: Pino.t => unit = %raw(`function (logger) {
-    console.log = function() {
-      var args = Array.from(arguments);
+open Belt
 
-      logger.uinfo(args.length > 1 ? args : args[0])
-    };
-    console.info = console.log; // Use uinfo for both console.log and console.info
+type chainData = {
+  chainId: float,
+  poweredByHyperSync: bool,
+  firstEventBlockNumber: option<int>,
+  latestProcessedBlock: option<int>,
+  timestampCaughtUpToHeadOrEndblock: option<Js.Date.t>,
+  numEventsProcessed: int,
+  latestFetchedBlockNumber: int,
+  currentBlockHeight: int,
+  numBatchesFetched: int,
+  endBlock: option<int>,
+  numAddresses: int,
+}
+@tag("status")
+type state =
+  | @as("disabled") Disabled({})
+  | @as("initializing") Initializing({})
+  | @as("active")
+  Active({
+      envioVersion: option<string>,
+      chains: array<chainData>,
+      indexerStartTime: Js.Date.t,
+      isPreRegisteringDynamicContracts: bool,
+      isUnorderedMultichainMode: bool,
+      rollbackOnReorg: bool,
+    })
 
-    console.debug = function() {
-      var args = Array.from(arguments);
+let chainDataSchema = S.schema((s): chainData => {
+  chainId: s.matches(S.float),
+  poweredByHyperSync: s.matches(S.bool),
+  firstEventBlockNumber: s.matches(S.option(S.int)),
+  latestProcessedBlock: s.matches(S.option(S.int)),
+  timestampCaughtUpToHeadOrEndblock: s.matches(S.option(S.datetime(S.string))),
+  numEventsProcessed: s.matches(S.int),
+  latestFetchedBlockNumber: s.matches(S.int),
+  currentBlockHeight: s.matches(S.int),
+  numBatchesFetched: s.matches(S.int),
+  endBlock: s.matches(S.option(S.int)),
+  numAddresses: s.matches(S.int),
+})
+let stateSchema = S.union([
+  S.literal(Disabled({})),
+  S.literal(Initializing({})),
+  S.schema(s => Active({
+    envioVersion: s.matches(S.option(S.string)),
+    chains: s.matches(S.array(chainDataSchema)),
+    indexerStartTime: s.matches(S.datetime(S.string)),
+    isPreRegisteringDynamicContracts: s.matches(S.bool),
+    isUnorderedMultichainMode: s.matches(S.bool),
+    rollbackOnReorg: s.matches(S.bool),
+  })),
+])
 
-      logger.udebug(args.length > 1 ? args : args[0]);
-    };
-    console.warn = function() {
-      var args = Array.from(arguments);
+// let setApiTokenEnv = {
+//   let initialted = ref(None)
+//   let envPath = NodeJs.Path.resolve([".env"])
+//   async apiToken => {
+//     // Execute once even with multiple calls
+//     if initialted.contents !== Some(apiToken) {
```

Comment:
Is this code still going to be used?

Reply:
I'd say yes, but in a little bit different form